ext/java/org/msgpack/jruby/Encoder.java

```diff
@@ -68,9 +71,13 @@ private void ensureRemainingCapacity(int c) {
   }
 
   private IRubyObject readRubyString() {
-    IRubyObject str = runtime.newString(new ByteList(buffer.array(), 0, buffer.position(), binaryEncoding, false));
-    buffer.clear();
-    return str;
+    if (streaming) {
+      return runtime.newString();
+    } else {
+      IRubyObject str = runtime.newString(new ByteList(buffer.array(), 0, buffer.position(), binaryEncoding, false));
+      buffer.clear();
+      return str;
+    }
```

Comment:
I want a code-comment here about why it will not read a String when "streaming".

Reply:
Comment added.